### PR TITLE
Use headers exchange in RabbitMQ.RPCClient

### PR DESCRIPTION
**Pull request by anronin**

### CODE REVIEW

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex [hunk 1]
```diff
@@ -48,48 +48,46 @@ defmodule MessageQueue.Adapters.RabbitMQ.RPCClient do
...
-  def handle_call({:exec, command}, from, %{channel: channel, queue: queue, calls: calls} = state) do
+  def handle_call(
+        {:exec, {service_name, _, _} = command},
+        from,
+        %{channel: channel, queue: queue, calls: calls} = state
+      ) do
     with {:ok, %{payload: payload, correlation_id: correlation_id}} <- Request.prepare(command),
          :ok <-
-           Basic.publish(channel, "", MessageQueue.rpc_queue(), payload,
+           Basic.publish(channel, "rpc", "", payload,
```
Comment (DmitryKK) [inline lib/message_queue/adapters/rabbitmq/rpc_client.ex:85]: explain this. Why did you remove `MessageQueue.rpc_queue()` and set headers only?

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex [hunk 2]
```diff
@@ -105,17 +103,21 @@ defmodule MessageQueue.Adapters.RabbitMQ.RPCClient do
...
 
       {from, calls} ->
         GenServer.reply(from, reply)
-        {:noreply, %{state | calls: calls}, :hibernate}
+        {:noreply, %{state | calls: calls}}
     end
   end
+
+  defp schedule_timeout_error(correlation_id) do
+    timeout = @call_timeout - :timer.seconds(5)
```
Comment (DmitryKK) [inline lib/message_queue/adapters/rabbitmq/rpc_client.ex:120]: why ?

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex (hunk 1 above)
Comment (DmitryKK) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:85]: if you remove rpc_queue then remove it from main module and config. and fix readme.md

Comment (anronin) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:85]: i think it's more flexible, we can bind different queues in one app, `GoogleHotelAds` and `GoogleMyHotelShop` for example. `MessageQueue.rpc_queue()` still used in RPCServer, we can remove it in global lib refactoring later if needed

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex (hunk 2 above)
Comment (anronin) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:120]: because timeout message must be returned before `call` will raise an timeout error, didn't find a way how to do it otherwise

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex [hunk 3]
```diff
@@ -105,17 +103,21 @@ defmodule MessageQueue.Adapters.RabbitMQ.RPCClient do
...
+    Process.send_after(__MODULE__, {:timeout, correlation_id}, timeout)
```
Comment (Judis) [inline lib/message_queue/adapters/rabbitmq/rpc_client.ex:121]: `Process.send_after` return `reference` as response, which can be used to cancel timer via `Process.cancel_timer/1` (https://hexdocs.pm/elixir/Process.html#cancel_timer/1).
May be we should add logic to do this if message is processed in expected time?

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex (hunk 1 above)
Comment (DmitryKK) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:85]: why do you want use different queues? it should be common way for rpc call. if we complicate the logic, it is more difficult for us to adapt

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex (hunk 3 above)
Comment (DmitryKK) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:121]: I agree with @judis. You can catch `call` timeout and remove `reply` timeout timer

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex (hunk 1 above)
Comment (anronin) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:85]: i don't see how it's complicate the logic and why we should use one queue for different channels implementations, so one channel will be wait while the other is being processed?

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex (hunk 3 above)
Comment (anronin) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:121]: not sure i understand @DmitryKK, Judis suggest not to `catch` the call but use cancel_timer after reply and this is not really necessary since `correlation_id` will be popped from state and thus `{:noreply, state}` will be returned when `{:timeout, correlation_id}` msg will come, but this can be done to reduce the number of unnecessary messages. I don't really like to use `catch` since it's expensive operation and do not provide any benefit here

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex (hunk 1 above)
Comment (DmitryKK) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:85]: rpc_server has `prefetch_count` is 1. channel will be wait in any case. More complex logic with headers it is more complex logic for another adapter. We can use different adapters for rpc and producer/consumer in future for example. It complicates adaptation

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex (hunk 3 above)
Comment (DmitryKK) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:121]: benefits - 1) you haven't to use time lag with 5 seconds 2.) caller will get correct error without exit 3.) we can add pretty log

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex (hunk 1 above)
Comment (anronin) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:85]: i dont use rpc_server, it's impossible to use it in current implementation, but that way i can use different rpc_receivers for different channel implementation. How it's more complex to use header with `{channel_name, true}` then queue with `rpc_app_name`? please take a look how it used now 
https://github.com/ChannexIO/server/pull/1150/files#diff-2e11df8fa4e02cc8b873b934d24e70cfc34a0e9d2df31d53c1ad61123c3b7c48

https://github.com/ChannexIO/server/pull/1150/files#diff-02518cd46a1c3653776a8825987d57406d8fe38eee1ff4aceb08646a864fb125R6

to use different adapters for rpc and producer/consumer in future we need even more refactoring for lib

File: lib/message_queue/adapters/rabbitmq/rpc_client.ex (hunk 3 above)
Comment (anronin) [reply lib/message_queue/adapters/rabbitmq/rpc_client.ex:121]: there is no time lag, `{:error, timeout}` will be returned after 30 seconds before call is exit and caller will get the same `{:error, timeout}` response as if use `catch`. What kind of info we can get in log except `service_name, function, args`?

**Review by Judis — APPROVED** [review]

**Review by DmitryKK — APPROVED** [review]
